### PR TITLE
Use Tanstack data to determine if addresses have assets

### PR DESCRIPTION
**Pull request by nop33**
My goal is to remove the API related properties from the `Address` type so that we only get them from Tanstack. One of the API data is `balances`. To remove `balances` I had to remove the `addressHasAssets` util function because it uses `address.balance`. This, in effect, led to all the changes in this PR.

To introduce clean new code, I had to refactor the AddressSelect related code so that it accepts address hashes as properties instead of the whole Address type.

EDIT:
Audit fails for a package with no patched version, so let's ignore.
```
│ Package             │ micromatch                                             │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Vulnerable versions │ <=4.0.7                                                │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Patched versions    │ <0.0.0                                                 │
```

### CODE REVIEW

File: apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes copy 2.tsx
Comment (mvaivre) [inline apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes copy 2.tsx:1]: 🤔

File: apps/desktop-wallet/src/features/addressFiltering/addressFilteringHooks.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import { useAddressesUnlistedFungibleTokens } from '@/api/addressesUnlistedTokensHooks'
+import { addressAlphBalanceQuery, addressTokensBalanceQuery } from '@/api/addressQueries'
+import { useAddressesLastTransactionHashes } from '@/api/addressTransactionsDataHooks'
+import { useFungibleTokenList } from '@/api/fungibleTokenListDataHooks'
+import { useAppSelector } from '@/hooks/redux'
+import { selectAllAddresses, selectAllAddressHashes } from '@/storage/addresses/addressesSelectors'
+import { isDefined } from '@/utils/misc'
+
+export const useFilteredAddressHashes = (text = '', hideEmpty?: boolean) => {
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/addressFiltering/addressFilteringHooks.tsx:32]: Correct. I will rework it with a clear head. Thanks for the feedback!

File: apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes copy 2.tsx
Comment (nop33) [reply apps/desktop-wallet/src/features/addressFiltering/useFilteredAddressHashes copy 2.tsx:1]: Deleted. Sorry!

File: apps/desktop-wallet/src/features/addressFiltering/addressFilteringHooks.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    allAddresses,
+    fungibleTokenList,
+    text,
+    unlistedFungibleTokens
+  ])
+
+  return filteredAddressHashes
+}
+
+export const useFilterEmptyAddresses = () => {
```
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/addressFiltering/addressFilteringHooks.tsx:111]: Renamed to `useAddressesWithBalance`

File: apps/desktop-wallet/src/features/addressFiltering/addressFilteringHooks.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nop33) [reply apps/desktop-wallet/src/features/addressFiltering/addressFilteringHooks.tsx:32]: Fixed

**Comment by nop33** [comment]
All your comments were addressed!

**Review by mvaivre — APPROVED** [review]

**Pull request merged**